### PR TITLE
Added HRRR method for 2-m diagnostics, added several diagnostics variables to the history files

### PR DESCRIPTION
**Pull request by tanyasmirnova**
The changes in this PR affect the FV3_HRRR physics suite. They include:
1. Flux-based method for 2-m diagnostics (sfc_diag.f). This diagnostics will be enabled when MYNN surface layer
    scheme and RUC LSM are used in the physics suite.
2. Several variables are added to the output (sfc_drv_ruc.F90). They
    include such variables as variable-density snow accumulation, accumulated snow melt,
    latent heat flux due to snow phase changes, sublimation, direct evaporation, canopy evaporation and
    transpiration fluxes. Fixed units of runoff variables.
3. Cleaned-up print statements. Introduced a test point with lat/lon for debugging in sfc_diag.f, sfc_drv_ruc.F90 and
    module_sf_ruclsm.F90.
4. Several modifications to the snow model in module_sf_ruclsm.F90.

### CODE REVIEW

**Review by mdtoyNOAA — APPROVED** [review]

File: physics/sfc_diag.f [hunk 1]
```diff
@@ -75,23 +95,131 @@ subroutine sfc_diag_run                                           &
...
+            else                   !  for dew formation, use saturated q at tskin
+              qss    = fpvs(tskin(i))
+              qss    = eps * qss/(ps(i) + epsm1 * qss)
+              q2m(i) = qss*wrk + max(qmin,q1c)*fhi
+            endif
+          else
+          !-- RUC lsm
+            if(chs.lt.1.E-5) then
+            !-- under very stable conditions use first level temperature
+              t2m(i) = t1(i)
```
Comment (joeolson42) [inline physics/sfc_diag.f]: I can only approve this PR if we never use this "use first model level" hack. I'd prefer it if this option was deleted.

File: physics/sfc_diag.f [hunk 2]
```diff
@@ -75,23 +95,131 @@ subroutine sfc_diag_run                                           &
...
+            else
+              q2m(i) = qsurf(i)*wrk + max(qmin,q1c)*fhi
+            endif
+          endif ! RUC lsm
+
+        else
+        !-- flux method
+          if(chs2(i).lt.1.E-5) then
+          !-- under very stable conditions use first level temperature
+            t2m(i) = t1(i)
```
Comment (joeolson42) [inline physics/sfc_diag.f]: same here. I thought this was proven to be a bad idea that resulted in speckling. This should be delete.

**Review by hannahcbarnes — APPROVED** [review]

File: physics/sfc_diag.f (hunk 1 above)
Comment (tanyasmirnova) [reply physics/sfc_diag.f]: @joeolson42 Joe, This is exactly what we have in HRRR 2-m diagnostics. We should discuss if we want to remove this option in RRFS at the RRFS meeting.

Comment (tanyasmirnova) [reply physics/sfc_diag.f]: @joeolson42 I removed the lines in question and repeated my test - the results were identical. Thus, it is a very rare condition, and most likely does not have any noticeable impact on 2-m diagnostics. Plus, it is used only with RUC LSM. I hope we could merge this PR as soon as possible as it contains some urgent changes for RRFS.

Comment (joeolson42) [reply physics/sfc_diag.f]: I'm glad to hear the condition where those lines are applied is rare, but I will only be satisfied if they are never applied. Plus, I don't think hacks like that will make it into the community repository, so they should probably be cut out now. Can you update the PR with those lines removed?

Comment (tanyasmirnova) [reply physics/sfc_diag.f]: @joeolson42 @SamuelTrahanNOAA Joe, you are strongly opposing the algorithm which is used in HRRR and also which is available to the WRF community. Also, the 2-m diagnostics in this PR is designed to be used **only** in the FV3_HRRR suite. I do not expect any objections from the community, when there is no impact on their work. In my CCPP work I haven't experienced any objections to the code that has impact only on RUC LSM. At this point any other change will need another regression test and will delay other important RRFS_dev changes in this PR. 
@SamuelTrahanNOAA I hope this PR could go to RRFS_dev as soon as possible.

Comment (joeolson42) [reply physics/sfc_diag.f]: It was always a bad idea. Using it in HRRRv4 doesn't justify it. We've often seen a warm bias in 2-m temperatures at night and this hack can only contribute to it.  Why cling to bad ideas?

Comment (tanyasmirnova) [reply physics/sfc_diag.f]: Obviously it is not perfect, but I can't agree that it is a bad idea. When there is no mixing in the surface layer, the atm. temperature near the surface is changing very little. Thus, for these conditions it is more likely to get a better estimate of 2-m temperature by simply setting it to the 1st level (8-m) temperature, than by computing it from the gradient between the skin temperature and 8-m temperature. To me, the former idea is better than the latter.

Comment (joeolson42) [reply physics/sfc_diag.f]: But this hack will always result in a high estimate. Anyone who knows stable boundary layer knows that stratification continues beyond the first 2 meters of the atmosphere. In many cases, with stagnant winds, the temperature difference between the surface and ~ 10 m (approx height of the first model level) can easily exceed 10 C. Even if 80% of the temperature difference is concentrated below 2 m, that still leaves a non-negligible difference between 2 m and the first model level. Using this hack, guarantees a high estimate. It's more than imperfect, it's misleading -- a blatant attempt to hide the old nighttime cold-bias problem. I'm sorry, but his hack should go. Our code should be better than this.

Comment (joeolson42) [reply physics/sfc_diag.f]: You make the commit and I'll help run the reg tests.

**Comment by SamuelTrahanNOAA** [comment]
Bad news: this PR breaks restart support: rap_restart and rap_sfcdiff_restart both fail. This is true even if I revert Sam Trahan's changes.

**Review by SamuelTrahanNOAA — APPROVED** [review]

**Pull request merged**